app/services/return-requirements/check-licence-ended.service.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+/**
+ * Fetches and checks data needed for the view `/return-requirements/{sessionId}/check-your-answers` page
+ * @module CheckLicenceService
+ */
+
+const LicenceModel = require('../../models/licence.model.js')
+
+/**
+ * Fetch the matching licence and return data needed for the check your answers page
+ *
+ * Was built to provide the data needed for the '/return-requirements/{id}/check-your-answers' page
+ *
+ * @param {string} id The UUID for the licence to fetch
+ *
+ * @returns {Promise<Object>} the data needed for "check your answers" page when user submits for approval
+ */
+async function go (id) {
+  const licence = await _fetchLicence(id)
+  const licenceEnded = _checkLicenceEnded(licence)
+  return licenceEnded
+}
+
+async function _fetchLicence (id) {
+  return LicenceModel.query()
+    .findById(id)
+    .select([
+      'expiredDate',
+      'id',
+      'lapsedDate',
+      'licenceRef',
+      'revokedDate',
+      'startDate'
+    ])
+}
+
+/**
+ * Check if a licence has ended.
+ * If it's not within range, we assume it's either lapsed, expired or revoked
+ * It returns true if the licence has ended, false otherwise.
+ *
+ * @param {object} licenceData - The licence data object
+ * @returns {boolean} - True if the licence has ended, false otherwise
+ */
+function _checkLicenceEnded (licence) {
+  const { $ends: ends } = licence
+
+  if (!ends) {
+    return false
+  }
+
+  const endDate = new Date(ends.date)
+  const today = new Date()
+  today.setHours(0, 0, 0, 0)
+
+  return endDate <= today
+}
```

Comment:
As `$ends()` is a function rather than a property we wouldn't normally use destructuring to invoke the method.

And the object it returns (when there is an end date) contains a date. Did some testing and we shouldn't need to create a new date instance.

So, I believe after these changes you'll get the same result.

```suggestion
  const ends = licence.$ends()

  if (!ends) {
    return false
  }

  const today = new Date()
  today.setHours(0, 0, 0, 0)

  return ends.date <= today
}
```